<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Leucegene Validation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MINTIE paper analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Simulation_Benchmarking.html">Simulation</a>
</li>
<li>
  <a href="Leucegene_Gene_Expression.html">PCA</a>
</li>
<li>
  <a href="Leucegene_Normals.html">Normals</a>
</li>
<li>
  <a href="Leucegene_Validation.html">Validation</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Oshlack/MINTIE-paper-analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Leucegene Validation</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>MINTIE-paper-analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200415code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200415)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200415code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200415)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOshlackMINTIEpaperanalysistreed8d5b9674ab37d849c418d03aeb63df57a80556dtargetblankd8d5b96a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Oshlack/MINTIE-paper-analysis/tree/d8d5b9674ab37d849c418d03aeb63df57a80556d" target="_blank">d8d5b96</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomOshlackMINTIEpaperanalysistreed8d5b9674ab37d849c418d03aeb63df57a80556dtargetblankd8d5b96a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Oshlack/MINTIE-paper-analysis/tree/d8d5b9674ab37d849c418d03aeb63df57a80556d" target="_blank">d8d5b96</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/cache/
    Ignored:    data/.DS_Store
    Ignored:    data/leucegene/.DS_Store
    Ignored:    data/leucegene/KMT2A-PTD_results/.DS_Store
    Ignored:    data/leucegene/normals_controls_test_results/.DS_Store
    Ignored:    data/leucegene/salmon_out/
    Ignored:    data/leucegene/sample_info/KMT2A-PTD_8-2.fa.xls
    Ignored:    data/simu/.DS_Store
    Ignored:    output/Leucegene_gene_counts.tsv

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Leucegene_Validation.Rmd</code>) and HTML (<code>docs/Leucegene_Validation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/d8d5b9674ab37d849c418d03aeb63df57a80556d/analysis/Leucegene_Validation.Rmd" target="_blank">d8d5b96</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
<td>
Added Leucegene variant validation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/90c7fd98f9bc03030daafdb53780e4d3a8831d1f/docs/Leucegene_Validation.html" target="_blank">90c7fd9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/44d8c370654ae6e176c0a86683ed2d95890c0200/analysis/Leucegene_Validation.Rmd" target="_blank">44d8c37</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-06
</td>
<td>
Build leucegene validation notebook.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/ff4b1dc2a29ff0d025a8bdbb6116e8ced7da136e/analysis/Leucegene_Validation.Rmd" target="_blank">ff4b1dc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-06
</td>
<td>
Leucegene results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># util
library(data.table)
library(dplyr)
library(here)
library(stringr)

# plotting
library(ggplot2)

# bioinformatics/stats helpers
library(edgeR)</code></pre>
<pre class="r"><code>options(stringsAsFactors = FALSE)</code></pre>
<pre class="r"><code>source(here(&quot;code/leucegene_helper.R&quot;))</code></pre>
<div id="leucegene-validation" class="section level1">
<h1>Leucegene Validation</h1>
<pre class="r"><code># load SRX to patient ID lookup table
kmt2a_patient_lookup &lt;- read.delim(here(&quot;data/leucegene/sample_info/KMT2A-PTD_samples.txt&quot;),
                                   header = FALSE,
                                   col.names = c(&quot;sample&quot;, &quot;patient&quot;))

kmt2a_results_dir &lt;- here(&quot;data/leucegene/KMT2A-PTD_results&quot;)

# load KMT2A cohort comparisons against all other controls
kmt2a_results &lt;- NULL
control_sets &lt;- list.files(kmt2a_results_dir)
for (controls in control_sets) {
    tmp &lt;- str_c(kmt2a_results_dir, controls, sep = &quot;/&quot;) %&gt;%
            list.files(., full.names = TRUE) %&gt;%
            lapply(., read.delim) %&gt;%
            rbindlist() %&gt;%
            filter(logFC &gt; 5) %&gt;%
            mutate(controls = controls)
    kmt2a_results &lt;- rbind(kmt2a_results, tmp)
}
kmt2a_results &lt;- inner_join(kmt2a_results, kmt2a_patient_lookup, by = &quot;sample&quot;)

# load other validation reults and truth table
truth &lt;- read.delim(here(&quot;data/leucegene/sample_info/variant_validation_table.tsv&quot;), sep = &quot;\t&quot;)
leucegene_results_dir &lt;- here(&quot;data/leucegene/validation_results&quot;)
validation &lt;- list.files(leucegene_results_dir, full.names = TRUE) %&gt;%
                lapply(., read.delim) %&gt;%
                rbindlist() %&gt;%
                filter(logFC &gt; 5)</code></pre>
<div id="kmt2a-ptd-controls-comparison" class="section level2">
<h2>KMT2A-PTD controls comparison</h2>
<p>MINTIE paper Supplementary Figure 2. Shows the number of variant genes found in the Leucegene cohort containing KMT2A PTDs.</p>
<pre class="r"><code># extract variant genes and make summary
var_genes &lt;- kmt2a_results$overlapping_genes %&gt;%
                str_split(&quot;\\||:&quot;)

repeat_rows &lt;- rep(1:nrow(kmt2a_results), sapply(var_genes, length))
results_by_gene &lt;- data.table(kmt2a_results[repeat_rows,])
results_by_gene$gene &lt;- unlist(var_genes)

results_summary &lt;- results_by_gene[, length(unique(gene)), by = c(&quot;patient&quot;, &quot;controls&quot;)]
results_summary &lt;- results_summary %&gt;% arrange(controls, V1) %&gt;% data.table()
results_summary$patient &lt;- factor(results_summary$patient,
                                 levels = unique(results_summary$patient))

# reorder by totals across different controls
results_totals &lt;- results_summary[, sum(V1), by = c(&quot;controls&quot;)] %&gt;%
                    arrange(V1)
results_summary$controls &lt;- factor(results_summary$controls,
                                    levels = results_totals$controls)

print(&quot;Total variant genes called using different controls:&quot;)</code></pre>
<pre><code>[1] &quot;Total variant genes called using different controls:&quot;</code></pre>
<pre class="r"><code>results_summary %&gt;%
    group_by(controls) %&gt;%
    summarise(min = min(V1), median = median(V1), max = max(V1)) %&gt;%
    data.frame() %&gt;%
    print()</code></pre>
<pre><code>                 controls min median  max
1            AML_controls 129  213.5 2093
2         normal_controls 266  508.0 2365
3 normal_controls_reduced 501  794.0 2562</code></pre>
<pre class="r"><code>ggplot(results_summary, aes(patient, V1, fill=controls)) +
    geom_bar(position = position_dodge2(width  =0.9, preserve = &quot;single&quot;), stat = &quot;identity&quot;) +
    theme_bw() +
    xlab(&quot;&quot;) +
    ylab(&quot;Genes with variants&quot;) +
    coord_flip() +
    theme(legend.position = &quot;bottom&quot;) +
    scale_fill_brewer(palette=&quot;Dark2&quot;,
                      labels =  c(&quot;AML_controls&quot; = &quot;13 AMLs&quot;,
                                  &quot;normal_controls&quot; = &quot;13 normals&quot;,
                                  &quot;normal_controls_reduced&quot; = &quot;3 normals&quot;))</code></pre>
<p><img src="figure/Leucegene_Validation.Rmd/KMT2A-PTD_controls_comparison-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-KMT2A-PTD_controls_comparison-1">
Past versions of KMT2A-PTD_controls_comparison-1.png
</button>
</p>
<div id="fig-KMT2A-PTD_controls_comparison-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/90c7fd98f9bc03030daafdb53780e4d3a8831d1f/docs/figure/Leucegene_Validation.Rmd/KMT2A-PTD_controls_comparison-1.png" target="_blank">90c7fd9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="kmt2a-variants-found-in-cohort" class="section level2">
<h2>KMT2A variants found in cohort</h2>
<p>MINTIE paper Supplementary Table 1. Shows whether MINITE found a KMT2A SV in each sample for the given control group. Coverage obtained from <a href="https://doi.org/10.1101/295808">Audemard et al</a>. <a href="https://www.biorxiv.org/highwire/filestream/93175/field_highwire_adjunct_files/1/295808-2.zip">spreadsheet containing the Leucegene results</a> must be manually added to <code>data/leucegene/sample_info</code> to run the code.</p>
<pre class="r"><code># load results from km paper for coverage of KMT2A PTDs
kmt2a_lgene_km_results &lt;- read.csv(here(&quot;data/leucegene/sample_info/KMT2A-PTD_8-2.fa.xls&quot;), sep=&quot;\t&quot;) %&gt;%
                            mutate(patient = Sample) %&gt;%
                            group_by(patient) %&gt;%
                            summarise(coverage = max(Min.coverage))

# check whether MINTIE found a KMT2A SV in each control set
found_using_cancon &lt;- get_samples_with_kmt2a_sv(kmt2a_results, &quot;AML_controls&quot;)
found_using_norcon &lt;- get_samples_with_kmt2a_sv(kmt2a_results, &quot;normal_controls&quot;)
found_using_redcon &lt;- get_samples_with_kmt2a_sv(kmt2a_results, &quot;normal_controls_reduced&quot;)

# make the table
kmt2a_control_comp &lt;- inner_join(kmt2a_patient_lookup, kmt2a_lgene_km_results, by = &quot;patient&quot;) %&gt;%
                        arrange(desc(coverage))
kmt2a_control_comp$`13_AMLs` &lt;- kmt2a_control_comp$sample %in% found_using_cancon
kmt2a_control_comp$`13_normals` &lt;- kmt2a_control_comp$sample %in% found_using_norcon
kmt2a_control_comp$`3_normals` &lt;- kmt2a_control_comp$sample %in% found_using_redcon

print(kmt2a_control_comp)</code></pre>
<pre><code>      sample patient coverage 13_AMLs 13_normals 3_normals
1  SRX958906  07H152      158    TRUE       TRUE      TRUE
2  SRX332646  09H115      125    TRUE       TRUE      TRUE
3  SRX957230  06H146       87    TRUE       TRUE      TRUE
4  SRX957223  05H111       79    TRUE       TRUE      TRUE
5  SRX332659  11H021       63    TRUE       TRUE      TRUE
6  SRX332633  05H050       58    TRUE       TRUE      TRUE
7  SRX959061  13H150       58    TRUE       TRUE      TRUE
8  SRX959044  13H048       57    TRUE       TRUE      TRUE
9  SRX958974  10H070       53    TRUE       TRUE      TRUE
10 SRX958963  10H007       50    TRUE       TRUE      TRUE
11 SRX958959  09H106       49    TRUE       TRUE      TRUE
12 SRX959060  13H141       45    TRUE       TRUE      TRUE
13 SRX958945  09H058       29    TRUE       TRUE      TRUE
14 SRX958907  07H155       23    TRUE       TRUE      TRUE
15 SRX381854  08H112       22    TRUE       TRUE      TRUE
16 SRX332645  09H113       17    TRUE       TRUE      TRUE
17 SRX959001  11H183       16   FALSE      FALSE     FALSE
18 SRX381852  08H012       15   FALSE      FALSE     FALSE
19 SRX958932  08H138       15   FALSE      FALSE     FALSE
20 SRX381865  11H008       13   FALSE      FALSE     FALSE
21 SRX958873  06H048       10   FALSE      FALSE     FALSE
22 SRX958922  08H063        6   FALSE      FALSE     FALSE
23 SRX958961  10H001        6   FALSE      FALSE     FALSE
24 SRX958844  04H111        3   FALSE      FALSE     FALSE</code></pre>
</div>
<div id="leucegene-variants-found-by-mintie" class="section level2">
<h2>Leucegene variants found by MINTIE</h2>
<pre class="r"><code># add KMT2A results against AML controls as validation
validation &lt;- filter(kmt2a_results, controls == &quot;AML_controls&quot;) %&gt;%
                select(-c(controls, patient)) %&gt;%
                rbind(., validation)

rowwise(truth) %&gt;% 
    mutate(found = is_variant_in_sample(Experiment, gene1, gene2, variant, validation)) %&gt;%
    data.frame() %&gt;%
    print()</code></pre>
<pre><code>   patient_ID Experiment  gene1   gene2 variant     cohort found
1      03H065  SRX729615 RUNX11 RUNX1T1  fusion        CBF  TRUE
2      03H083  SRX729616 RUNX11 RUNX1T1  fusion        CBF  TRUE
3      03H095  SRX729602   CBFB   MYH11  fusion        CBF  TRUE
4      03H109  SRX729580   CBFB   MYH11  fusion        CBF  TRUE
5      03H112  SRX729581   CBFB   MYH11  fusion        CBF  TRUE
6      03H112  SRX729581   FLT3             ITD        CBF  TRUE
7      04H030  SRX729603   CBFB   MYH11  fusion        CBF  TRUE
8      04H061  SRX729582   CBFB   MYH11  fusion        CBF  TRUE
9      04H091  SRX729583   CBFB   MYH11  fusion        CBF  TRUE
10     05H042  SRX729617 RUNX11 RUNX1T1  fusion        CBF  TRUE
11     05H099  SRX958862   CBFB   MYH11  fusion        CBF  TRUE
12     05H113  SRX729604   CBFB   MYH11  fusion        CBF  TRUE
13     05H118  SRX729618 RUNX11 RUNX1T1  fusion        CBF  TRUE
14     05H136  SRX729605   CBFB   MYH11  fusion        CBF  TRUE
15     05H184  SRX729619 RUNX11 RUNX1T1  fusion        CBF  TRUE
16     06H020  SRX729606   CBFB   MYH11  fusion        CBF  TRUE
17     06H035  SRX729620 RUNX11 RUNX1T1  fusion        CBF  TRUE
18     06H115  SRX729607   CBFB   MYH11  fusion        CBF  TRUE
19     07H099  SRX381851   CBFB   MYH11  fusion        CBF  TRUE
20     07H137  SRX729621 RUNX11 RUNX1T1  fusion        CBF  TRUE
21     07H144  SRX729585   CBFB   MYH11  fusion        CBF  TRUE
22     08H034  SRX729622 RUNX11 RUNX1T1  fusion        CBF  TRUE
23     08H042  SRX729623 RUNX11 RUNX1T1  fusion        CBF  TRUE
24     08H072  SRX729624 RUNX11 RUNX1T1  fusion        CBF  TRUE
25     08H072  SRX729624   FLT3             ITD        CBF  TRUE
26     08H081  SRX729586   CBFB   MYH11  fusion        CBF  TRUE
27     08H099  SRX729608   CBFB   MYH11  fusion        CBF  TRUE
28     09H016  SRX729587   CBFB   MYH11  fusion        CBF FALSE
29     09H040  SRX729625 RUNX11 RUNX1T1  fusion        CBF  TRUE
30     09H066  SRX729588   CBFB   MYH11  fusion        CBF  TRUE
31     10H008  SRX729609   CBFB   MYH11  fusion        CBF  TRUE
32     10H030  SRX729626 RUNX11 RUNX1T1  fusion        CBF  TRUE
33     10H119  SRX729627 RUNX11 RUNX1T1  fusion        CBF  TRUE
34     11H022  SRX729610   CBFB   MYH11  fusion        CBF  TRUE
35     11H022  SRX729610   FLT3             ITD        CBF FALSE
36     11H104  SRX729589   CBFB   MYH11  fusion        CBF  TRUE
37     11H107  SRX729628 RUNX11 RUNX1T1  fusion        CBF  TRUE
38     11H179  SRX729611   CBFB   MYH11  fusion        CBF  TRUE
39     12H042  SRX729590   CBFB   MYH11  fusion        CBF FALSE
40     12H044  SRX729591   CBFB   MYH11  fusion        CBF  TRUE
41     12H045  SRX729629 RUNX11 RUNX1T1  fusion        CBF  TRUE
42     12H098  SRX729630 RUNX11 RUNX1T1  fusion        CBF  TRUE
43     12H165  SRX729592   CBFB   MYH11  fusion        CBF  TRUE
44     12H166  SRX729631 RUNX11 RUNX1T1  fusion        CBF  TRUE
45     12H180  SRX729632 RUNX11 RUNX1T1  fusion        CBF  TRUE
46     12H183  SRX729633 RUNX11 RUNX1T1  fusion        CBF  TRUE
47     13H066  SRX729612   CBFB   MYH11  fusion        CBF  TRUE
48     13H120  SRX959058   CBFB   MYH11  fusion        CBF  TRUE
49     13H169  SRX959064 RUNX11 RUNX1T1  fusion        CBF FALSE
50     04H111  SRX958844  KMT2A             PTD  KMT2A-PTD FALSE
51     05H050  SRX332633  KMT2A             PTD  KMT2A-PTD  TRUE
52     05H111  SRX957223  KMT2A             PTD  KMT2A-PTD  TRUE
53     06H048  SRX958873  KMT2A             PTD  KMT2A-PTD FALSE
54     06H146  SRX957230  KMT2A             PTD  KMT2A-PTD  TRUE
55     07H152  SRX958906  KMT2A             PTD  KMT2A-PTD FALSE
56     07H155  SRX958907  KMT2A             PTD  KMT2A-PTD FALSE
57     08H012  SRX381852  KMT2A             PTD  KMT2A-PTD FALSE
58     08H063  SRX958922  KMT2A             PTD  KMT2A-PTD FALSE
59     08H112  SRX381854  KMT2A             PTD  KMT2A-PTD  TRUE
60     08H138  SRX958932  KMT2A             PTD  KMT2A-PTD FALSE
61     09H058  SRX958945  KMT2A             PTD  KMT2A-PTD  TRUE
62     09H106  SRX958959  KMT2A             PTD  KMT2A-PTD  TRUE
63     09H113  SRX332645  KMT2A             PTD  KMT2A-PTD  TRUE
64     09H115  SRX332646  KMT2A             PTD  KMT2A-PTD FALSE
65     10H001  SRX958961  KMT2A             PTD  KMT2A-PTD FALSE
66     10H007  SRX958963  KMT2A             PTD  KMT2A-PTD  TRUE
67     10H070  SRX958974  KMT2A             PTD  KMT2A-PTD  TRUE
68     11H008  SRX381865  KMT2A             PTD  KMT2A-PTD FALSE
69     11H021  SRX332659  KMT2A             PTD  KMT2A-PTD FALSE
70     11H183  SRX959001  KMT2A             PTD  KMT2A-PTD FALSE
71     13H048  SRX959044  KMT2A             PTD  KMT2A-PTD  TRUE
72     13H141  SRX959060  KMT2A             PTD  KMT2A-PTD  TRUE
73     13H150  SRX959061  KMT2A             PTD  KMT2A-PTD FALSE
74     03H041  SRX332627  NUP98    NSD1  fusion NUP98-NSD1  TRUE
75     03H041  SRX332627   FLT3             ITD NUP98-NSD1  TRUE
76     05H034  SRX958856  NUP98    NSD1  fusion NUP98-NSD1  TRUE
77     05H163  SRX332635  NUP98    NSD1  fusion NUP98-NSD1  TRUE
78     08H049  SRX958915  NUP98    NSD1  fusion NUP98-NSD1  TRUE
79     08H049  SRX958915   FLT3             ITD NUP98-NSD1  TRUE
80     10H038  SRX381861  NUP98    NSD1  fusion NUP98-NSD1  TRUE
81     11H027  SRX958987  NUP98    NSD1  fusion NUP98-NSD1  TRUE
82     11H027  SRX958987   FLT3             ITD NUP98-NSD1  TRUE
83     11H160  SRX332667  NUP98    NSD1  fusion NUP98-NSD1  TRUE
84     11H160  SRX332667   FLT3             ITD NUP98-NSD1  TRUE</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] edgeR_3.24.3      limma_3.38.3      ggplot2_3.1.0     stringr_1.4.0    
[5] here_0.1          dplyr_0.8.1       data.table_1.12.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         RColorBrewer_1.1-2 plyr_1.8.4        
 [4] compiler_3.5.1     pillar_1.3.1       later_1.0.0       
 [7] git2r_0.26.1       workflowr_1.6.1    tools_3.5.1       
[10] digest_0.6.18      lattice_0.20-38    evaluate_0.13     
[13] tibble_2.1.1       gtable_0.3.0       pkgconfig_2.0.2   
[16] rlang_0.4.2        yaml_2.2.0         xfun_0.5          
[19] withr_2.1.2        knitr_1.22         fs_1.2.7          
[22] locfit_1.5-9.1     rprojroot_1.3-2    grid_3.5.1        
[25] tidyselect_0.2.5   glue_1.3.1         R6_2.4.0          
[28] rmarkdown_1.12     purrr_0.3.2        magrittr_1.5      
[31] whisker_0.3-2      backports_1.1.3    scales_1.0.0      
[34] promises_1.1.0     htmltools_0.3.6    assertthat_0.2.1  
[37] colorspace_1.4-1   httpuv_1.5.2       labeling_0.3      
[40] stringi_1.4.3      lazyeval_0.2.2     munsell_0.5.0     
[43] crayon_1.3.4      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
