<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Leucegene Normals</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MINTIE paper analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Simulation_Benchmarking.html">Simulation</a>
</li>
<li>
  <a href="Leucegene_Gene_Expression.html">PCA</a>
</li>
<li>
  <a href="Leucegene_Normals.html">Normals</a>
</li>
<li>
  <a href="Leucegene_Validation.html">Validation</a>
</li>
<li>
  <a href="RCH_B-ALL.html">B-ALLs</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Oshlack/MINTIE-paper-analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Leucegene Normals</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-05-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>MINTIE-paper-analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200415code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200415)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200415code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200415)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOshlackMINTIEpaperanalysistreef504dcbd4605181dbfccdc3775de172d237652b0targetblankf504dcba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Oshlack/MINTIE-paper-analysis/tree/f504dcbd4605181dbfccdc3775de172d237652b0" target="_blank">f504dcb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomOshlackMINTIEpaperanalysistreef504dcbd4605181dbfccdc3775de172d237652b0targetblankf504dcba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Oshlack/MINTIE-paper-analysis/tree/f504dcbd4605181dbfccdc3775de172d237652b0" target="_blank">f504dcb</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/cache/
    Ignored:    data/.DS_Store
    Ignored:    data/RCH_B-ALL/
    Ignored:    data/leucegene/.DS_Store
    Ignored:    data/leucegene/KMT2A-PTD_results/.DS_Store
    Ignored:    data/leucegene/normals_controls_test_results/.DS_Store
    Ignored:    data/leucegene/salmon_out/
    Ignored:    data/leucegene/sample_info/KMT2A-PTD_8-2.fa.xls
    Ignored:    data/simu/.DS_Store
    Ignored:    output/Leucegene_gene_counts.tsv

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Leucegene_Normals.Rmd</code>) and HTML (<code>docs/Leucegene_Normals.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/a6006888da26ba20e8599f360d3fc640cd54488a/docs/Leucegene_Normals.html" target="_blank">a600688</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/1c40e33ed980b6e6f7dfdf3290618aae2781e5e0/docs/Leucegene_Normals.html" target="_blank">1c40e33</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/bbc278a2ce8cc4777cf358cd4d3552461341a4dd/analysis/Leucegene_Normals.Rmd" target="_blank">bbc278a</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
<td>
Refactoring
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/87b4e62442a254ec41b1ffdd6b55824a14a9463e/docs/Leucegene_Normals.html" target="_blank">87b4e62</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/af503f253494b562986aef0e6a9e18b5d17d22c6/analysis/Leucegene_Normals.Rmd" target="_blank">af503f2</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
<td>
Refactoring
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/5c045b534e7c10361889255bbefe79111089f01a/docs/Leucegene_Normals.html" target="_blank">5c045b5</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/90c7fd98f9bc03030daafdb53780e4d3a8831d1f/docs/Leucegene_Normals.html" target="_blank">90c7fd9</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/ff4b1dc2a29ff0d025a8bdbb6116e8ced7da136e/analysis/Leucegene_Normals.Rmd" target="_blank">ff4b1dc</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-06
</td>
<td>
Leucegene results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/358aa53d68acc377f4eca81bd919c2cc2b9b3f72/docs/Leucegene_Normals.html" target="_blank">358aa53</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/453d754398ac058d75d1b096c6217d25c028549f/analysis/Leucegene_Normals.Rmd" target="_blank">453d754</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-04
</td>
<td>
Added controls comparison in normals analysis. Added variant class collation function. Added variant summary for
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Oshlack/MINTIE-paper-analysis/4a5d6aeedc2d6f903b0cebb8fddee018a73682e8/docs/Leucegene_Normals.html" target="_blank">4a5d6ae</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/9556ebb67fcee8884d57646f4a86dfbd02f42a61/analysis/Leucegene_Normals.Rmd" target="_blank">9556ebb</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-01
</td>
<td>
Added leucegene normals analysis. Added expressed genes analysis to leucegene gene expression analysis.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># util
library(data.table)
library(dplyr)
library(here)
library(stringr)

# plotting
library(ggplot2)</code></pre>
<pre class="r"><code>options(stringsAsFactors = FALSE)</code></pre>
<pre class="r"><code>source(here(&quot;code/leucegene_helper.R&quot;))</code></pre>
<div id="leucegene-normals" class="section level1">
<h1>Leucegene Normals</h1>
<p>Here we generate the results presented in the MINTIE paper, of the method run on a set of non-cancer samples obtained from Leucegene.</p>
<pre class="r"><code># load MINTIE results from leucegene normals
normals_results_dir &lt;- here(&quot;data/leucegene/normals_results&quot;)
normals_results &lt;- list.files(normals_results_dir, full.names = TRUE) %&gt;% 
                        lapply(., read.delim) %&gt;%
                        rbindlist() %&gt;%
                        filter(logFC &gt; 5)

# load cell type info and add to results
celltype &lt;- read.delim(here(&quot;data/leucegene/sample_info/celltypes_info.tsv&quot;))
normals_results &lt;- inner_join(normals_results, celltype,
                              by = c(&quot;sample&quot; = &quot;SRX_ID&quot;))</code></pre>
<div id="variant-summary" class="section level2">
<h2>Variant Summary</h2>
<p>Summary results for variants called by MINTIE on Leucegene normals.</p>
<pre class="r"><code>summary_by_sample &lt;- normals_results %&gt;%
                        group_by(sample) %&gt;%
                        summarise(variants = length(unique(variant_id)))
summary_by_class &lt;- collate_vartypes(normals_results) %&gt;%
                        group_by(class) %&gt;%
                        summarise(variants = length(unique(variant_id))) %&gt;%
                        mutate(fraction = variants / sum(variants))
print(summary_by_sample)</code></pre>
<pre><code># A tibble: 23 x 2
   sample    variants
   &lt;chr&gt;        &lt;int&gt;
 1 SRX372044      181
 2 SRX372045      137
 3 SRX372046      213
 4 SRX372047      639
 5 SRX372048     1119
 6 SRX372049      143
 7 SRX372050      497
 8 SRX372051      449
 9 SRX372052      235
10 SRX372053       75
# … with 13 more rows</code></pre>
<pre class="r"><code>print(summary_by_class)</code></pre>
<pre><code># A tibble: 4 x 3
  class                          variants fraction
  &lt;chr&gt;                             &lt;int&gt;    &lt;dbl&gt;
1 Fusion                               69   0.0117
2 Novel splice variant               1992   0.338 
3 Transcribed structural variant     1234   0.210 
4 Unknown                            2593   0.440 </code></pre>
</div>
<div id="variant-genes" class="section level2">
<h2>Variant Genes</h2>
<p>MINTIE paper Figure 4 showing the number of variant genes called across the Leucegene normal samples.</p>
<pre class="r"><code>results_summary &lt;- get_results_summary(mutate(normals_results, group_var = cell_type),
                                       group_var_name = &quot;cell_type&quot;)

ggplot(results_summary, aes(cell_type, V1, group=sample)) + 
    geom_bar(position = position_dodge2(width = 0.9, preserve = &quot;single&quot;), stat = &quot;identity&quot;) +
    theme_bw() +
    xlab(&quot;&quot;) +
    ylab(&quot;Genes with variants&quot;)</code></pre>
<p><img src="figure/Leucegene_Normals.Rmd/plot_variant_genes-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_variant_genes-1">
Past versions of plot_variant_genes-1.png
</button>
</p>
<div id="fig-plot_variant_genes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/1c40e33ed980b6e6f7dfdf3290618aae2781e5e0/docs/figure/Leucegene_Normals.Rmd/plot_variant_genes-1.png" target="_blank">1c40e33</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/4a5d6aeedc2d6f903b0cebb8fddee018a73682e8/docs/figure/Leucegene_Normals.Rmd/plot_variant_genes-1.png" target="_blank">4a5d6ae</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="library-size-and-variant-number-correlation" class="section level2">
<h2>Library Size and Variant Number Correlation</h2>
<p>Perform correlation calculation on the library size and number of variant genes found per sample.</p>
<p><a href="Leucegene_Gene_Expression.html">Leucegene Gene Expression</a> notebook must be run before this chunk to generate the expression counts matrix.</p>
<pre class="r"><code># load counts data, calculate library sizes and add to results summary
counts &lt;- fread(here(&quot;output/Leucegene_gene_counts.tsv&quot;))
libsizes &lt;- apply(counts, 2, sum) %&gt;% data.frame()
colnames(libsizes) &lt;- &quot;libsize&quot;
libsizes$sample &lt;- factor(rownames(libsizes),
                          levels = results_summary$sample)
results_summary &lt;- left_join(results_summary, libsizes, by =&quot;sample&quot;, &quot;sample&quot;)

lib_var_cor &lt;- cor(results_summary$libsize, results_summary$V1, method = &quot;spearman&quot;)
print(paste(&quot;Spearman correlation between library size and variant genes called:&quot;, lib_var_cor))</code></pre>
<pre><code>[1] &quot;Spearman correlation between library size and variant genes called: 0.140845074722306&quot;</code></pre>
<pre class="r"><code>ggplot(results_summary, aes(libsize, V1, colour = cell_type)) +
    geom_point() +
    theme_bw() +
    ylab(&quot;Genes with variants&quot;)</code></pre>
<p><img src="figure/Leucegene_Normals.Rmd/libsize_var_cor-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-libsize_var_cor-1">
Past versions of libsize_var_cor-1.png
</button>
</p>
<div id="fig-libsize_var_cor-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/1c40e33ed980b6e6f7dfdf3290618aae2781e5e0/docs/figure/Leucegene_Normals.Rmd/libsize_var_cor-1.png" target="_blank">1c40e33</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/358aa53d68acc377f4eca81bd919c2cc2b9b3f72/docs/figure/Leucegene_Normals.Rmd/libsize_var_cor-1.png" target="_blank">358aa53</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="protein-coding-genes" class="section level2">
<h2>Protein Coding Genes</h2>
<p>Proportion of protein coding genes observed in the MINTIE results.</p>
<pre class="r"><code># load CHESS gene reference containing gene types
chess_genes &lt;- read.delim(gzfile(here(&quot;data/ref/chess2.2.genes.gz&quot;)))

# join gene info with results and summarise by gene type
results_by_gene &lt;- get_results_by_gene(normals_results)
gene_count &lt;- left_join(results_by_gene, chess_genes, by = c(&quot;gene&quot; = &quot;Gene_Name&quot;)) %&gt;%
                group_by(Gene_Type) %&gt;%
                summarise(n_genes = length(unique(gene))) %&gt;%
                data.table()

n_protein_coding &lt;- gene_count[gene_count$Gene_Type == &quot;protein_coding&quot;, &quot;n_genes&quot;]
print(paste(&quot;proportion of protein coding genes =&quot;, n_protein_coding / sum(gene_count$n_genes)))</code></pre>
<pre><code>[1] &quot;proportion of protein coding genes = 0.81547619047619&quot;</code></pre>
</div>
<div id="controls-comparison" class="section level2">
<h2>Controls Comparison</h2>
<p>MINTIE Supplementary Figure 3 showing variant genes called in Leucegene Total White Blood Cell samples with different cell types as control groups.</p>
<pre class="r"><code># get TWBC results
controls_comp &lt;- normals_results[normals_results$cell_type == &quot;Total white blood cells&quot;,]
controls_comp$controls &lt;- &quot;twbc&quot;
controls_comp$cell_type &lt;- NULL

# load comparisons against all other controls
controls_test_dir &lt;- here(&quot;data/leucegene/normals_controls_test_results&quot;)
controls_comp &lt;- load_controls_comparison(controls_test_dir) %&gt;%
                    rbind(controls_comp, .)
results_summary &lt;- get_results_summary(mutate(controls_comp,
                                              group_var = controls),
                                       group_var_name = &quot;controls&quot;)

print(&quot;Total variant genes called using different controls:&quot;)</code></pre>
<pre><code>[1] &quot;Total variant genes called using different controls:&quot;</code></pre>
<pre class="r"><code>print(results_summary)</code></pre>
<pre><code>      sample controls  V1
1  SRX372045       bc 345
2  SRX372044       bc 365
3  SRX372046       bc 443
4  SRX372045     gran 130
5  SRX372046     gran 247
6  SRX372044     gran 279
7  SRX372045     mono 138
8  SRX372044     mono 187
9  SRX372046     mono 192
10 SRX372044       tc 329
11 SRX372045       tc 378
12 SRX372046       tc 419
13 SRX372045     twbc 156
14 SRX372044     twbc 193
15 SRX372046     twbc 233</code></pre>
<pre class="r"><code>ggplot(results_summary, aes(sample, V1, fill=controls)) +
    geom_bar(position=position_dodge2(width=0.9, preserve=&quot;single&quot;), stat=&quot;identity&quot;) +
    theme_bw() +
    xlab(&quot;&quot;) +
    ylab(&quot;Genes with variants&quot;) +
    scale_fill_brewer(palette = &quot;RdYlBu&quot;,
                      labels =  c(&quot;mono&quot; = &quot;Monocytes&quot;,
                                  &quot;twbc&quot; = &quot;Total white blood cells&quot;,
                                  &quot;gran&quot; = &quot;Granulocytes&quot;,
                                  &quot;tc&quot; = &quot;T-Cells&quot;,
                                  &quot;bc&quot; = &quot;B-Cells&quot;))</code></pre>
<p><img src="figure/Leucegene_Normals.Rmd/controls_comparison-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-controls_comparison-1">
Past versions of controls_comparison-1.png
</button>
</p>
<div id="fig-controls_comparison-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/87b4e62442a254ec41b1ffdd6b55824a14a9463e/docs/figure/Leucegene_Normals.Rmd/controls_comparison-1.png" target="_blank">87b4e62</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Oshlack/MINTIE-paper-analysis/blob/358aa53d68acc377f4eca81bd919c2cc2b9b3f72/docs/figure/Leucegene_Normals.Rmd/controls_comparison-1.png" target="_blank">358aa53</a>
</td>
<td>
Marek Cmero
</td>
<td>
2020-05-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.1.0     stringr_1.4.0     here_0.1          dplyr_0.8.1      
[5] data.table_1.12.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1         RColorBrewer_1.1-2 plyr_1.8.4        
 [4] compiler_3.5.1     pillar_1.3.1       later_1.0.0       
 [7] git2r_0.26.1       workflowr_1.6.1    tools_3.5.1       
[10] digest_0.6.18      evaluate_0.13      tibble_2.1.1      
[13] gtable_0.3.0       pkgconfig_2.0.2    rlang_0.4.2       
[16] cli_1.1.0          yaml_2.2.0         xfun_0.5          
[19] withr_2.1.2        knitr_1.22         fs_1.2.7          
[22] rprojroot_1.3-2    grid_3.5.1         tidyselect_0.2.5  
[25] glue_1.3.1         R6_2.4.0           fansi_0.4.0       
[28] rmarkdown_1.12     purrr_0.3.2        magrittr_1.5      
[31] whisker_0.3-2      backports_1.1.3    scales_1.0.0      
[34] promises_1.1.0     htmltools_0.3.6    assertthat_0.2.1  
[37] colorspace_1.4-1   httpuv_1.5.2       labeling_0.3      
[40] utf8_1.1.4         stringi_1.4.3      lazyeval_0.2.2    
[43] munsell_0.5.0      crayon_1.3.4      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
