---
title: "Leucegene Normals"
output:
  workflowr::wflow_html:
      code_folding: hide
---

```{r knitr, include = FALSE}
DOCNAME = "Leucegene_Normals"
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.path     = paste0("cache/", DOCNAME, "/"),
                      cache.comments = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 9,
                      fig.height     = 5,
                      dev            = c("png", "pdf"),
                      message        = FALSE,
                      warning        = FALSE)
```


```{r libraries, cache = FALSE}
# util
library(data.table)
library(dplyr)
library(here)
library(stringr)

# plotting/tables
library(ggplot2)
library(gt)
```


```{r options, cache = FALSE}
options(stringsAsFactors = FALSE)
```


```{r source, cache = FALSE}
source(here("code/leucegene_helper.R"))
```

Leucegene Normals
=================

Here we generate the results presented in the MINTIE paper, of the method
run on a set of non-cancer samples obtained from Leucegene.

```{r load_data}
# load MINTIE results from leucegene normals
normals_results_dir <- here("data/leucegene/normals_results")
normals_results <- list.files(normals_results_dir, full.names = TRUE) %>% 
                        lapply(., read.delim) %>%
                        rbindlist(fill = TRUE) %>%
                        filter(logFC > 5)

# load cell type info and add to results
celltype <- read.delim(here("data/leucegene/sample_info/celltypes_info.tsv"))
normals_results <- inner_join(normals_results, celltype,
                              by = c("sample" = "SRX_ID"))
```

## Variant Summary

Summary results for variants called by MINTIE on Leucegene normals.

```{r}
normals_results %>%
    group_by(sample) %>%
    summarise(variants = length(unique(variant_id))) %>%
    summarise(min = min(variants),
              median = median(variants),
              max = max(variants)) %>%
    gt() %>%
    tab_header(
        title = md("**Variants called by sample summary**")
    ) %>%
    tab_options(
        table.font.size = 12
    ) %>%
    cols_label(
        min = md("**Min**"),
        median = md("**Median**"),
        max = md("**Max**")
    )

collate_vartypes(normals_results) %>%
    group_by(class) %>%
    summarise(variants = length(unique(variant_id))) %>%
    mutate(fraction = variants / sum(variants)) %>%
    gt() %>%
    fmt_number(columns = vars(fraction), decimals = 3) %>%
    tab_header(
        title = md("**Variants called summary by class**")
    ) %>%
    tab_options(
        table.font.size = 12
    ) %>%
    cols_label(
        variants = md("**Variants**"),
        fraction = md("**Fraction**")
    )
```

## Variant Genes

MINTIE paper Figure 4B showing the number of variant genes called across
the Leucegene normal samples.

```{r plot_variant_genes, fig.width=9, fig.height=4}
results_summary <- get_results_summary(mutate(normals_results, group_var = cell_type),
                                       group_var_name = "cell_type")

results_summary %>%
    summarise(min = min(V1),
              median = median(V1),
              max = max(V1),
              total = sum(V1)) %>%
    gt() %>%
    tab_header(
        title = md("**Variant genes called by sample summary**")
    ) %>%
    tab_options(
        table.font.size = 12
    ) %>%
    cols_label(
        min = md("**Min**"),
        median = md("**Median**"),
        max = md("**Max**"),
        total = md("**Total**")
    )

ggplot(results_summary, aes(cell_type, V1, group=sample)) + 
    geom_bar(position = position_dodge2(width = 0.9, preserve = "single"), stat = "identity") +
    theme_bw() +
    theme(axis.text.x = element_text(size = 11),
      axis.text.y = element_text(size = 11)) +
    xlab("") +
    ylab("Genes with variants")
```

## Library Size and Variant Number Correlation

Perform correlation calculation on the library size and number of variant
genes found per sample.

[Leucegene Gene Expression](Leucegene_Gene_Expression.html) notebook must be
run before this chunk to generate the expression counts matrix.

```{r libsize_var_cor}
# load counts data, calculate library sizes and add to results summary
counts <- fread(here("output/Leucegene_gene_counts.tsv"))
libsizes <- apply(counts, 2, sum) %>% data.frame()
colnames(libsizes) <- "libsize"
libsizes$sample <- factor(rownames(libsizes),
                          levels = results_summary$sample)
results_summary <- left_join(results_summary, libsizes, by ="sample", "sample")

lib_var_cor <- cor(results_summary$libsize, results_summary$V1, method = "spearman")
print(paste("Spearman correlation between library size and variant genes called:", lib_var_cor))
ggplot(results_summary, aes(libsize, V1, colour = cell_type)) +
    geom_point() +
    theme_bw() +
    ylab("Genes with variants")
```

## Protein Coding Genes

Proportion of protein coding genes observed in the MINTIE results.

```{r protein_coding_proportion}
# load CHESS gene reference containing gene types
chess_genes <- get_chess_genes(gzfile(here("data/ref/chess2.2.genes.gz")))

# join gene info with results and summarise by gene type
results_by_gene <- get_results_by_gene(normals_results)
gene_count <- left_join(results_by_gene, chess_genes, by = "gene") %>%
                group_by(Gene_Type) %>%
                summarise(n_genes = length(unique(gene))) %>%
                data.table()

n_protein_coding <- gene_count[gene_count$Gene_Type == "protein_coding", "n_genes"]
n_var_genes <- sum(gene_count$n_genes)
paste("proportion of protein coding genes =",
    (n_protein_coding / n_var_genes) %>% round(4),
    paste0("(", n_protein_coding, "/", n_var_genes, ")")) %>%
    print()
```

## Controls Comparison

MINTIE Supplementary Figure 4 showing variant genes called in Leucegene
Total White Blood Cell samples with different cell types as control groups.

```{r controls_comparison}
# get TWBC results
controls_comp <- normals_results[normals_results$cell_type == "Total white blood cells",]
controls_comp$controls <- "twbc"
controls_comp$cell_type <- NULL

# load comparisons against all other controls
controls_test_dir <- here("data/leucegene/normals_controls_test_results")
controls_comp <- load_controls_comparison(controls_test_dir) %>%
                    rbind(controls_comp, ., fill = TRUE)
results_summary <- get_results_summary(mutate(controls_comp,
                                              group_var = controls),
                                       group_var_name = "controls")

results_summary %>%
    group_by(controls) %>%
    summarise(tcount = sum(V1)) %>%
    gt() %>%
    tab_header(
        title = md("**Total variant genes called using different controls**")
    ) %>%
    tab_options(
        table.font.size = 12
    ) %>%
    cols_label(
        controls = md("**Controls**"),
        tcount = md("**Variant genes**")
    )

ggplot(results_summary, aes(sample, V1, fill=controls)) +
    geom_bar(position=position_dodge2(width=0.9, preserve="single"), stat="identity") +
    theme_bw() +
    xlab("") +
    ylab("Genes with variants") +
    scale_fill_brewer(palette = "RdYlBu",
                      labels =  c("mono" = "Monocytes",
                                  "twbc" = "Total white blood cells",
                                  "gran" = "Granulocytes",
                                  "tc" = "T-Cells",
                                  "bc" = "B-Cells"))
```

### 1..N controls experiments

MINTIE Figure 4C showing range of variant genes called
in granulocytes with different combinations of 1..N controls. The
table output also includes numbers for the runs where 0 controls
were used.

```{r n_controls_comparison, fig.width=9, fig.height=4.5}
# load in data
ncon_dirs <- here("data/leucegene/normals_ncontrols_test_results/")
ncon_results <- data.frame()
for (ncon_dir in list.dirs(ncon_dirs, recursive = FALSE)) {
    run <- str_split(ncon_dir, "ncon")[[1]] %>%
                tail(1)
    ncon <- str_split(run, "r") %>% unlist() %>% head(1) %>% as.numeric()
    results <- list.files(ncon_dir, pattern = "*results.tsv.gz", full.names = TRUE) %>%
                        lapply(., read.delim) %>%
                        rbindlist() %>%
                        get_results_by_gene() %>%
                        mutate(n_controls = ncon, run = run)
    ncon_results <- rbindlist(list(ncon_results, results), fill = TRUE)
}
ncon_results <- ncon_results[logFC > 5 | is.na(logFC),]

# prepare results summary
results_summary <- get_results_summary(mutate(ncon_results,
                                              group_var = run),
                                       group_var_name = "run")
results_summary$n_controls <- results_summary$run %>%
                                as.character() %>%
                                str_split(., pattern = "r") %>%
                                lapply(., head, n=1) %>% unlist() %>% as.numeric()
rs <- results_summary %>%
        group_by(sample, n_controls) %>%
        summarise(mean=round(mean(V1, na.rm=TRUE)),
                  min=min(V1, na.rm=TRUE),
                  max=max(V1, na.rm=TRUE)) %>%
        data.frame()

rs %>%
    gt() %>%
    tab_header(
        title = md("**Total variant genes called using different numbers of controls**")
    ) %>%
    tab_options(
        table.font.size = 12
    ) %>%
    cols_label(
        n_controls = md("**Controls**"),
        mean = md("**Mean**"),
        min = md("**Min**"),
        max = md("**Max**")
    )

rs$n_controls <- as.factor(rs$n_controls)
ggplot(rs[rs$n_controls!=0,], aes(sample, mean, fill=n_controls)) +
    geom_bar(position=position_dodge2(width=0.9, preserve="single"), stat="identity") +
    geom_errorbar(aes(ymin=min, ymax=max, group=n_controls), width=.2,
                 position=position_dodge(.9)) +
    theme_bw() +
    theme(axis.text.x = element_text(size = 11),
        axis.text.y = element_text(size = 11)) +
    xlab("") +
    ylab("Mean genes with variants") +
    scale_fill_brewer(palette = "Dark2") +
    theme(legend.position = "bottom")
```


