---
title: "Leucegene Gene Expression"
output:
  workflowr::wflow_html:
      code_folding: hide
---

```{r knitr, include = FALSE}
DOCNAME = "Leucegene_Gene_Expression"
knitr::opts_chunk$set(autodep        = TRUE,
                      cache          = FALSE,
                      cache.path     = paste0("cache/", DOCNAME, "/"),
                      cache.comments = FALSE,
                      echo           = TRUE,
                      error          = FALSE,
                      fig.align      = "center",
                      fig.width      = 9,
                      fig.height     = 5,
                      message        = FALSE,
                      warning        = FALSE)
```


```{r libraries, cache = FALSE}
# util
library(data.table)
library(dplyr)
library(here)
library(stringr)

# plotting/tables
library(ggplot2)
library(RColorBrewer)
library(gt)

# bioinformatics/stats helpers
library(tximport)
library(limma)
library(edgeR)
library(matrixStats)
```

```{r options, cache = FALSE}
options(stringsAsFactors = FALSE)
```

```{r source, cache = FALSE}
source(here("code/leucegene_helper.R"))
```

Leucegene Gene Expression
=========================

Here we generate a PCA plot for all the Leucegene samples used in the MINTIE
paper KMT2A-PTD and normal sample analyses, and perform a few small expression
analyses presented in the paper. Salmon output is not provided in the repository
and must be generated by the user.

```{r load_data, cache = FALSE}
salmon_dir <- here("data/leucegene/salmon_out")

# construct list of quant.sf files 
quant_files <- list.files(salmon_dir,
                          full.names = TRUE,
                          recursive = TRUE,
                          pattern = "quant.sf")

# transcript > gene reference file for CHESS
tx2gene <- read.delim(gzfile(here("data/ref/tx2gene.txt.gz")))

# import quant files
txi <- tximport(quant_files,
                type = "salmon",
                countsFromAbundance = "lengthScaledTPM",
                tx2gene = tx2gene,
                ignoreTxVersion = FALSE)

# load sample info
celltype <- read.delim(here("data/leucegene/sample_info/celltypes_info.tsv"))
kmt2a_samples <- read.delim(here("data/leucegene/sample_info/KMT2A-PTD_samples.txt"), header = FALSE)$V1
aml_controls <- read.delim(here("data/leucegene/sample_info/selected_13_CBF_AML_controls.txt"), header = FALSE)$V1
nup_samples <- read.delim(here("data/leucegene/sample_info/NUP98-NSD1_samples.txt"), header = FALSE)$V1

# reduced normal control set (used in KMT2A-PTD analysis)
s1 <- celltype$SRX_ID[celltype$cell_type == "Total white blood cells"][1]
s2 <- celltype$SRX_ID[celltype$cell_type == "Monocytes"][1]
s3 <- celltype$SRX_ID[celltype$cell_type == "Granulocytes"][1]
reduced_normal_controls <- c(s1, s2, s3)
```

## PCA plot

MINTIE paper Supplememtary Figure 1. PCA of gene expression derived
from Salmon quantification for Leucegene KMT2A-PTD and normals. Normal
samples used as controls in the reduced set of controls are circled.

```{r PCA_plot}
# get counts matrix
counts <- txi$counts
colnames(counts) <- list.files(salmon_dir)
write.table(counts, file = here("output/Leucegene_gene_counts.tsv"), sep = "\t", quote = FALSE, row.names = FALSE)

# variance stabilised, log2 CPM transformation
ve <- voom(counts)$E

# select 500 most variable genes
select <- order(rowVars(ve), decreasing = T)[1:500]

# perform PCA amd select first two components
pr <- prcomp(ve[select,])
pc <- data.frame(pr$rotation[,1:2])
pc$sample <- rownames(pc)

# attach labels
pc <- left_join(pc, celltype, by = c("sample" = "SRX_ID"))
pc$cell_type[pc$sample %in% kmt2a_samples] <- "KMT2A-PTD"
pc$cell_type[pc$sample %in% nup_samples] <- "NUP98-NSD1"
pc$cell_type[pc$sample %in% aml_controls] <- "CBF AML controls"
pc <- pc[!is.na(pc$cell_type),]

# make colour mappings
cols <- brewer.pal(8, "RdBu")
names(cols) <- c("KMT2A-PTD",
                 "CBF AML controls",
                 "Granulocytes",
                 "Monocytes",
                 "Total white blood cells",
                 "T-cells",
                 "B-cells")
pc$cell_type <- factor(pc$cell_type, levels = names(cols))

# plot
ggplot(pc, aes(PC1, PC2, colour = cell_type)) +
    geom_point(size = 2.5) +
    theme_bw() +
    scale_color_manual(values = cols) +
    theme(legend.title = element_blank()) +
    geom_point(data = pc[pc$sample %in% reduced_normal_controls,],
               shape = 1, size = 5, fill = NA, colour = 'darkgrey')
```

## Expressed genes

Number of expressed genes found in Leucegene normals and percentage of those
genes that are protein coding.

```{r expressed_genes}
# load CHESS gene reference containing gene types
chess_genes <- get_chess_genes(gzfile(here("data/ref/chess2.2.genes.gz")))

# construct a normal counts matrix of "expressed" counts
# (>1 CPM in at least one sample)
normal_counts <- counts[,colnames(counts) %in% celltype$SRX_ID]
keep <- rowSums(cpm(normal_counts) > 1) >= 1
normal_counts <- normal_counts[keep,]

# tally gene types
expressed_genes <- rownames(normal_counts)
gene_types <- filter(chess_genes, gene %in% expressed_genes) %>%
                group_by(Gene_Type) %>%
                summarise(gene_count = length(unique(GFF_ID)))
gene_types <- rbind(gene_types, c("Total", sum(gene_types$gene_count)))

# results table
data.frame(gene_types) %>%
    gt() %>%
    tab_header(
        title = md("**Gene classifications**")
    ) %>%
    tab_options(
        table.font.size = 12
    ) %>%
    cols_label(
        Gene_Type = md("**Gene type**"),
        gene_count = md("**Count**")
    )

n_protein_coding <- filter(gene_types, Gene_Type == "protein_coding") %>% select(gene_count)
print(paste("Proportion of expressed genes that are protein coding:",
            (as.numeric(n_protein_coding) / length(expressed_genes)) %>% round(4)))
```
